pgml-extension/src/bindings/transformers/transformers.py

```diff
@@ -983,3 +997,617 @@ def generate(model_id, data, config):
             )
             all_preds.extend(decoded_preds)
     return all_preds
+
+
+#######################
+# LLM Fine-Tuning
+#######################
+
+
+class PGMLCallback(TrainerCallback):
```

Comment:
I wouldn't be opposed to this functionality living in it's own file like `tune.py`, since transformers is getting a bit beefy.

Reply:
transformers.py is hardcoded in several places. Needs some more refactoring and testing to accomplish moving finetuning code to tune.py. Will revisit this in the next iteration. #1378